core/variables.js

```diff
@@ -285,25 +285,33 @@ Blockly.Variables.renameVariable = function(workspace, variable,
  * Prompt the user for a new variable name.
  * @param {string} promptText The string of the prompt.
  * @param {string} defaultText The default value to show in the prompt's field.
+ * @param {string} opt_type Optional type of variable, like 'string' or 'list'.
  * @param {function(?string)} callback A callback. It will be passed the new
  *     variable name, or null if the user picked something illegal.
  */
-Blockly.Variables.promptName = function(promptText, defaultText, callback) {
+Blockly.Variables.promptName = function(promptText, defaultText, opt_type, callback) {
   Blockly.prompt(promptText, defaultText, function(newVar) {
     // Merge runs of whitespace.  Strip leading and trailing whitespace.
     // Beyond this, all names are legal.
     if (newVar) {
-      newVar = newVar.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
-      if (newVar == Blockly.Msg.RENAME_VARIABLE ||
-          newVar == Blockly.Msg.NEW_VARIABLE) {
-        // Ok, not ALL names are legal...
-        newVar = null;
-      }
+      newVar = Blockly.Variables.validateName_(newVar, opt_type);
     }
     callback(newVar);
   });
 };
 
+Blockly.Variables.validateName_ = function(name, opt_type) {
+  if (!opt_type || !opt_type == Blockly.BROADCAST_MESSAGE_VARIABLE_TYPE) {
+    name = name.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
+    if (name == Blockly.Msg.RENAME_VARIABLE ||
```

Comment:
This means that a broadcast message can be named "Rename variable" or "new variable", even though no other variable type allows that.  Is that intentional?

Reply:
This is intentional. These names were allowed in 2.0, and setting these as broadcast message names doesn't cause conflicts. 
I'm also not entirely convinced that we need to disallow these cases from other types of variables, but I didn't want to mess with variables in this PR.

Reply:
sg